### PR TITLE
TE-1840 fixed outdated templates

### PR DESCRIPTION
**Pull request by stereomon**
- Developer(s): @stereomon

- Ticket: https://spryker.atlassian.net/browse/TE-1840

- Academy PR: ACADEMY_URL_HERE

- Release Group: URL_HERE

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Spryk                 | patch (currently 0.x)  |                       |

#### Release Notes

This patch will update some templates to be used for generating files. The composer.json now contains by default `spryker/kernel` in require, `spryker/code-sniffer` and `spryker/testify` in require-dev. The `codeception.yml` now uses the psr-4 namespace for PHPUnit. The generated *DependencyProvider contains now the `@method` annotation for the `getConfig()` method.

-----------------------------------------

##### Change log

Bugfixes

- Updated `composer.json.twig`.
- Updated `codeception.yml.twig`.
- Updated `ZedDependencyProvider.php.twig`.

### CODE REVIEW

File: config/spryk/templates/composer.json.twig [hunk 1]
```diff
@@ -3,7 +3,12 @@
   "description": "{{ module }} module",
   "license": "proprietary",
   "require": {
-    "php": ">=7.1"
+    "php": ">=7.1",
+    "spryker/kernel": "*"
```
Comment (dereuromark) [inline config/spryk/templates/composer.json.twig]: This might get flagged by tooling.
We cant use a real value?

Comment (stereomon) [reply config/spryk/templates/composer.json.twig]: We can't as this is a moving target and we can't know which one is the correct version at this time.

Comment (dereuromark) [reply config/spryk/templates/composer.json.twig]: Do you expect Kernel major to go up this often?
With this people will always have to adjust their modules after running spryks.. That is far less ideal than us maintaining the major here (which we can even automate using a post-hook for Kernel majoring :smile: )

File: src/SprykerSdk/Spryk/Model/Spryk/Definition/Builder/SprykDefinitionBuilderInterface.php [hunk 2]
```diff
@@ -16,14 +15,14 @@ interface SprykDefinitionBuilderInterface
      * @param string $sprykName
      * @param array|null $preDefinedDefinition
      *
-     * @return \SprykerSdk\Spryk\Model\Spryk\Definition\SprykDefinitionInterface
+     * @return $this
      */
-    public function buildDefinition(string $sprykName, ?array $preDefinedDefinition = null): SprykDefinitionInterface;
+    public function buildDefinition(string $sprykName, ?array $preDefinedDefinition = null): self;
```
Comment (dereuromark) [inline src/SprykerSdk/Spryk/Model/Spryk/Definition/Builder/SprykDefinitionBuilderInterface.php]: Is this chaining? Then please no :self here

**Comment by dereuromark** [comment]
The template ideally uses code fencing for the files in changelog.

**Comment by dereuromark** [comment]
Can we also add the two missing files phpstan.neon and phpstan.json into https://github.com/spryker-sdk/spryk/blob/master/config/spryk/templates/gitattributes.twig ? Can also be a follow up though.

File: tests/_data/src/Pyz/Zed/FooBar/Communication/Controller/IndexController.php [hunk 3]
```diff
@@ -15,6 +15,7 @@
  */
 class IndexController extends AbstractController
 {
+
```
Comment (dereuromark) [inline tests/_data/src/Pyz/Zed/FooBar/Communication/Controller/IndexController.php]: That looks like a mistake

**Comment by dereuromark** [comment]
I fixed the code fencing for `@method` in your descr. Now all good to go.

**Pull request merged**